cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/engine/sql/capability/DefaultPullCapability.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.engine.sql.capability;
+
+public enum DefaultPullCapability implements PullCapability {
```

Comment:
Do you need multiple enums? Does it make sense to have HADOOP_PULL and SPARK_RDD_PULL part of same enum?

Reply:
I want to separate them as to keep Spark in the Spark packages. This way, if we were to implement support for a new engine, or deprecate an existing engine, we don't need to update this enum all the time.

Reply:
I've moved all values into the same class.